src/common/hooks/useDefaultProvider.ts

```diff
@@ -0,0 +1,13 @@
+import { useMemo } from "react";
+import { ethers, providers } from "ethers";
+import { NETWORK_NAME } from "../../config";
+
+export const useDefaultProvider = (): { provider: providers.Provider } => {
```

Comment:
I'm wondering if it's worth adding a fake signer doing nothing ?Otherwise consumer will have to handle the fact that both hook provider has a different signature and eventually no signer ?

Raising just from the point of view of the consumer. Don't need to add if it's ok as a consumer

Reply:
I dont really want to do that until i work on the write transactions. Had the idea to have one hook instead.

Reply:
Yeah I'm fine as look as for consumer it's not harder to use (like need to check signer is defined everywhere ? leave it open but probably nothing to do for the moment